lib/pure/includes/osenv.nim

```diff
@@ -201,27 +102,17 @@ else:
     ## * `existsEnv proc <#existsEnv,string>`_
     ## * `delEnv proc <#delEnv,string>`_
     ## * `envPairs iterator <#envPairs.i>`_
-
-    # Note: by storing the string in the environment sequence,
-    # we guarantee that we don't free the memory before the program
-    # ends (this is needed for POSIX compliance). It is also needed so that
-    # the process itself may access its modified environment variables!
     when nimvm:
       discard "built into the compiler"
     else:
-      var indx = findEnvVar(key)
-      if indx >= 0:
-        environment[indx] = key & '=' & val
-      else:
-        add environment, (key & '=' & val)
-        indx = high(environment)
       when defined(windows) and not defined(nimscript):
         when useWinUnicode:
           var k = newWideCString(key)
           var v = newWideCString(val)
           if setEnvironmentVariableW(k, v) == 0'i32: raiseOSError(osLastError())
         else:
-          if setEnvironmentVariableA(key, val) == 0'i32: raiseOSError(osLastError())
+          if setEnvironmentVariableA(key, val) == 0'i32:
+            raiseOSError(osLastError())
```

Comment:
```suggestion
            raiseOSError(osLastError(), $(key, val))
```

so that relevant context is shown on error instead of a generic error msg

liekwise in other cases

(no need for additional wording, `$(key, val)` has all the information needed to give the missing context)